src/imfilter.jl

```diff
@@ -25,4 +25,58 @@ function imfilter_gpu(img, krn)
   real.(result[CartesianIndices(finalsize)]) |> Array
 end
 
-const imfilter_kernel = CUDA.functional() ? imfilter_gpu : imfilter_cpu
+function imfilter_opencl(img, krn)
+  # retrieve basic info
+  N = ndims(img)
+  T = ComplexF64
+
+  # retrieve OpenCL info
+  device, ctx, queue = OpenCL.cl.create_compute_context()
+
+  # build OpenCL program kernels
+  conj_kernel = build_conj_kernel(ctx)
+  mult_kernel = build_mult_kernel(ctx)
+
+  # pad krn to common size with img
+  padkrn = zeros(eltype(img), size(img))
+  padkrn[CartesianIndices(krn)] = krn
+
+  # convert to Complex
+  fftimg = T.(img)
+  fftkrn = T.(padkrn)
+
+  # OpenCl setup
+  plan = CLFFT.Plan(T, ctx, size(img))
```

Comment:
You are not loading CLFFT.jl yet, so this code is not compiling.

Reply:
fixed